### ISSUE TITLE
[$250] Invoice - Invoice title page shows not here page

### ISSUE THREAD

**Issue by izarutskaya** [Expensify/App#47003]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.17-0
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**Email or phone of affected tester (no customers):** biruknew45+2293dh@gmail.com
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Issue reported by:** Applause-Internal team

## Action Performed:
1. Go to [staging.new.expensify.com].
2. Click on the floating action button (FAB).
3. Click on "new workspace" at the bottom.
4. It takes you to the workspace editor. Navigate to the settings page by clicking on the back button at the top left corner only once.
5. Stay on the settings page and click on the FAB.
6. Select "send invoice."
7. Enter the amount.
8. Enter the user.
9. Verify that the "send from" workspace is the same as the workspace you created.
10. Click on "send invoice."
11. It takes you to the invoice chat. Open the invoice preview. (You might need to open another chat and come back to the invoice chat to see the preview.)
12. Click on the header.
13. Click on the title.

## Expected Result:
The title page should open.

## Actual Result:
A "hmm... it's not here" page appears, and both the "back" button and the "go back to home page" link are not responsive

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [x] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

https://github.com/user-attachments/assets/364d7056-5963-49fd-a929-9686324ef84d

![Bug6554644_1722082038003!1](https://github.com/user-attachments/assets/b5f866ea-ee2e-45af-b77d-2318875fdbe8)

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01d7c703877e9d58e3</li>
        <li>Upwork Job ID: 1823549296018846731</li>
        <li>Last Price Increase: 2024-09-18</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @eVoloshchak</details>

**Comment by zinzaducnm** [comment]
## Proposal
### Please re-state the problem that we are trying to solve in this issue.
A "hmm... it's not here" page appears, and both the "back" button and the "go back to home page" link are not responsive

### What is the root cause of that problem?
```
/** Report fields attached to the policy */
fieldList?: Record<string, OnyxCommon.OnyxValueWithOfflineFeedback<PolicyReportField>>;
```
`fieldList` data is defined as an object. But after creating workspace, `fieldList` in `respoonse.policy` is array. I can see it in console log as below
```
Finished API request in 750ms - {"command":"CreateWorkspace","jsonCode":200,"requestID":"8b1d5cfa4c3584ed-HKG"}
{
    "approvalMode": "OPTIONAL",
    ...
    "fieldList": [
        {
            "defaultExternalID": null,
            "defaultValue": "{report:type} {report:startdate}",
            "deletable": true,
            "disabledOptions": [],
            "externalID": null,
            "externalIDs": [],
            "fieldID": "text_title",
            "isTax": false,
            "keys": [],
            "name": "title",
            "orderWeight": 0,
            "origin": null,
            "target": "expense",
            "type": "formula",
            "value": null,
            "values": []
        }
    ],
    ...
    "id": "F8AC0C1478D5ECBE",
    "ownerAccountID": 16529869
}
```

In WorkspaceProfilePage, there is an API request to get policy from BE (`Policy.openPolicyProfilePage(route.params.policyID);`) to update `fieldList` to right object. But this API cannot be called because queue is paused. 
https://github.com/Expensify/App/blob/66ae37403897f907d26e5dcb3b1b3744ea1055da/src/pages/workspace/WorkspaceProfilePage.tsx#L97
I find in logs that there are gaps between deferred updates. So `shouldPauseQueue` is set to true to ensure the client resolves the gap in onyx updates


After sending invoice to user and choosing send from the new workspace, an error occurred when visiting `EditReportFieldPage`. 
https://github.com/Expensify/App/blob/66ae37403897f907d26e5dcb3b1b3744ea1055da/src/pages/EditReportFieldPage.tsx#L53
This happens because `policy?.fieldList?` is an array and you cannot get right `reportField`. The `FullPageNotFoundView` is shown.
https://github.com/Expensify/App/blob/66ae37403897f907d26e5dcb3b1b3744ea1055da/src/pages/EditReportFieldPage.tsx#L60


### What changes do you think we should make in order to solve the problem?

- API request with command `CreateWorkspace`, should return the right type of fieldList that is defined in onyx type `fieldList?: Record<string, OnyxCommon.OnyxValueWithOfflineFeedback<PolicyReportField>>;`
- Find the the gap in onyx updates after creating workspace and fix it. Then `fieldList` is updated correctly after calling API request with command `OpenPolicyProfilePage`


### What alternative solutions did you explore? (Optional)
We can fix it tempolary at client side to get right `fieldList` by using this code to replace the code at line 53 of `EditReportFieldPage.tsx`:
https://github.com/Expensify/App/blob/66ae37403897f907d26e5dcb3b1b3744ea1055da/src/pages/EditReportFieldPage.tsx#L53
```
    const reportField = report?.fieldList?.[fieldKey] ?? (isArray(policy?.fieldList) ? policy?.fieldList?.find((o) = o.fieldID === fieldKey) : policy?.fieldList?.[fieldKey]);
```

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by eVoloshchak** [comment]
Cannot reproduce this issue either, @dominictb, @zinzaducnm, could you please double check if this is still reproducible for you?

**Comment by zinzaducnm** [comment]
@eVoloshchak I cannot reproduce now. I checked the log and saw that:
- After `CreateWorkspace` API called, `policy.fieldList` is array
- Then API request with {"command":"OpenPolicyProfilePage"} is returned with correct `fieldList` data

As I inspected it in my [proposal](https://github.com/Expensify/App/issues/47003#issuecomment-2283091981)

> In WorkspaceProfilePage, there is an API request to get policy from BE (Policy.openPolicyProfilePage(route.params.policyID);) to update fieldList to right object. But this API cannot be called because queue is paused.

It seems issue related to queue not happening anymore.

**Comment by anmurali** [comment]
Yes, I get an unexpected error in the send flow and then trying to change the title or description of the invoice triggers a its not here page error.

**Comment by anmurali** [comment]
@zinzaducnm - can you maybe retry and restate your proposal?

**Comment by eVoloshchak** [comment]
@dominictb, thank you for the [steps](https://github.com/Expensify/App/issues/47003#issuecomment-2314819164), was able to reproduce the issue

I noticed two problems when applying your [proposal](https://github.com/Expensify/App/issues/47003#issuecomment-2274645111) 

1. The title in the header and the title on the edit invoice title page are not the same
2. When you update the title, it isn't updated in the header


https://github.com/user-attachments/assets/eecea85b-6db2-487b-bf06-6e4a265f2b63

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (Second week)
